drivers/ethernet/nxp_enet/eth_nxp_enet.c

```diff
@@ -908,3 +955,23 @@ int nxp_enet_##n##_init(void)								\
 	SYS_INIT(nxp_enet_##n##_init, POST_KERNEL, 0);
 
 DT_INST_FOREACH_STATUS_OKAY(NXP_ENET_INIT)
+
+#undef DT_DRV_COMPAT
+#define DT_DRV_COMPAT nxp_enet1g
+
+#define NXP_ENET1G_INIT(n)								\
+											\
+int nxp_enet1g_##n##_init(void)								\
```

Comment:
This function looks the same as the one above, am I missing some difference?

If they are the same you can just call the DT for each macro twice, once for each compat, on the same function

Reply:
One function will expand to extracting "nxp,enet" compatible nodes data and the other to extracting "nxp,enet1g" compatible nodes data. I am not sure if it will work when DT_DRV_COMPAT will be always set to nxp_enet.

And if there will be both "nxp,enet" and "nxp,enet1g" compatible nodes with ok status, both will have instance number 0, won't they?

Reply:
You can #undef the DT_DRV_COMPAT before it and re-#define it or just don't used the INST macro and specify the compatible as an argument

Reply:
Done.